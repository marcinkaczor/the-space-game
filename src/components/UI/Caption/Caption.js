import React from 'react'
import PropTypes from 'prop-types'
import { Typography } from '@material-ui/core'

const Caption = ({ title = 'Loading ...' }) => (
  <Typography variant='caption'>{title}</Typography>
)

Caption.propTypes = {
  title: PropTypes.string
}

export default Caption
