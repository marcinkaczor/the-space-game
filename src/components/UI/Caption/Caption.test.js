/* global describe, test, it, expect */
import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import Caption from '.'

describe('Caption', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<Caption />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders with title', () => {
    const title = 'Loading test ...'

    const wrapper = mount(<Caption title={title} />)
    expect(wrapper.text()).toEqual(title)
  })
})
