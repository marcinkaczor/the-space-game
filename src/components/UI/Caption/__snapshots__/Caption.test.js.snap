// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Caption snapshot renders 1`] = `
<span
  className="MuiTypography-root MuiTypography-caption"
>
  Loading ...
</span>
`;
